### PR TITLE
Updated the tests to the new test method

### PR DESCRIPTION
**Pull request by AnnaDinaburgVulikh**
Hi @sleviim ,
It's the third part of the long pr.
This one includes only the ammended tests for lesson 3 : strins and variables.

### CODE REVIEW

File: docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py [hunk 1]
```diff
@@ -18,25 +19,35 @@
...
+         'check my_message variable definition'],
+        [r'''.*f["']My name is \{my_(full_)?name''' +
+         r'''\} and I'm from \{my_city_name\}''',
+         [r'''(.*['"]My name is\s*['"]\s*[+]\s*my_(full_)?name''' +
+          r'''\s*[+]\s*['"]\s*and I'm from\s*['"]\s*[+]\s*''' +
+          r'''my_city_name)|(.*['"]My name is \%s and I'm from \%s''' +
+          r'''['"]\s*\%\s*\(my_(full_)?name,\s*my_city_name\))''',
+          'a better definition for my_message should use f"string"',
+          'expected the use of f"string" for my_message variable, ' +
+          'make sure to include the previous variables']]
```
Comment (sleviim) [inline docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py:50]: I think a slight alignment is required.

File: docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py [hunk 2]
```diff
@@ -64,36 +75,38 @@ def test_times(helpers):
...
-         f'For the input: {helpers.input[0]},' +
-         'some of the expected text was missing']
+    error_message_for_partial_letter = (
+        f'For the input: {helpers.input[0]},' +
+        'some of the expected text was missing'
+        )
+
+    helpers.tests_list = [
+        [r'''\bprint\(f['|"]\{.*[date].*\}\\n\\tFor\\n\\t\{.*[name].*\}.*''',
+         'make sure to use \\t, \\n and f\' in your print statement.'],
```
Comment (sleviim) [inline docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py:88]: This one can be very tricky, as not all students know how to use tab (\t) or necessarily use newlines (\n).

File: docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py (hunk 1 above)
Comment (AnnaDinaburgVulikh) [reply docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py:50]: Lines 42-50 is one array/answer to be tested.
It tests for the best solution(42-43) and afterwords 2 other possible solutions(44-47).

File: docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py (hunk 2 above)
Comment (AnnaDinaburgVulikh) [reply docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py:88]: Both of those are included in the slides on page 5.4, that's why I tested for them. If you think another solution is in order, I am open for ideas.

Comment (sleviim) [reply docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py:88]: Maybe the test should check a regex for the tabs and new lines?
I think the text itself is more important here.

Comment (AnnaDinaburgVulikh) [reply docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py]: Hi @sleviim , updated the check as discussed.
Removed the text check, only the first line is checked ans the secondary solution checks for the tabs only.

File: docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py (hunk 1 above)
Comment (sleviim) [reply docs/course_materials/exercises/test_exercises/03_Variables_and_datatypes/test_homework_strings.py:50]: Ack

**Review by sleviim — APPROVED** [review]

**Pull request merged**